src/dotnet/APIView/APIViewWeb/Pages/Assemblies/Review.cshtml.cs

```diff
@@ -84,6 +87,27 @@ public async Task<IActionResult> OnGetAsync(string id, string revisionId = null)
                 showDocumentation: ShowDocumentation, showDiffOnly: ShowDiffOnly, diffContextSize: REVIEW_DIFF_CONTEXT_SIZE,
                 diffContextSeperator: DIFF_CONTEXT_SEPERATOR);
 
+            if (ReviewContent == default(ReviewContentModel))
+            {
+                // Check if you can get review from legacy data
+                var legacyReview = await _reviewManager.GetLegacyReviewAsync(User, id);
+                if (legacyReview != null)
+                {
+                    var legacyRevision = legacyReview.Revisions.FirstOrDefault(r =>
+                        !string.IsNullOrEmpty(r.Files[0].Language) && !string.IsNullOrEmpty(r.Files[0].PackageName));
+
+                    var review = await _reviewManager.GetReviewAsync(language: legacyRevision.Files[0].Language,
```

Comment:
This will fail if `legacyRevision` is null and that's possible for any older reviews.

Reply:
Well then it will be a legitimated failure won't it?

Reply:
Let's handle null pointer and return 404 in that case.